source/broadband/webpa_notification.c

```diff
@@ -282,6 +291,98 @@ void FactoryResetCloudSyncTask()
 		WalInfo("FactoryResetCloudSync Thread created Successfully\n");
 	}
 }
+void *SyncNotifyRetry()
+{
+	pthread_detach(pthread_self());
+	char *dbCMC = NULL;
+	int retryCount = 0;
+	int backoffRetryTime = 0;
+	int backoff_max_time = 9;
+	struct timespec ts;
+	int c = 3;
+	int  rv;    
+	int max_retry_sleep = (int) pow(2, backoff_max_time) -1;
+	WalInfo("max_retry_sleep is %d\n", max_retry_sleep );
+        
+	while(FOREVER())
+	{
+		if(backoffRetryTime < max_retry_sleep)
+       	{
+			backoffRetryTime = (int) pow(2, c) -1;
+		}
+		clock_gettime(CLOCK_REALTIME, &ts);
+		ts.tv_sec += backoffRetryTime;
+		//wait for backoff delay for retransmission
+		WalInfo("Wait for backoffRetryTime %d sec to receive cloud_conn_online event\n", backoffRetryTime);
+		rv = pthread_cond_timedwait(&sync_condition, &sync_mutex, &ts);
+		if (rv == 0)
+		{
+			WalInfo("Received connection online event signal for retrying sync notification.\n");
+		} 
+		else if (rv == ETIMEDOUT) 
+		{
+			WalPrint("SyncNotifyRetry thread: Timeout occurred.\n");
+		} 
+		else 
+		{
+			WalError("SyncNotifyRetry thread: Error occurred: %d\n", rv);	
+			continue;
+		}
+		dbCMC = getParameterValue(PARAM_CMC);
+		if(g_syncNotifyInProgress == 1)
+		{
+			WalInfo("dbCMC value is %s\n", dbCMC);			
+			WalInfo("PARAM_NOTIFY is in progress\n");
+			continue;
+		}
+		if(strcmp(dbCMC,"512"))
+		{       
+			//send sync notification to cloud for cloud and CPE sync
+			WalInfo("dbCMC value is %s\n", dbCMC);
+			WalInfo("Retrying sync notification as cloud and CPE are out of sync\n");
+			NotifyData *notifyData = (NotifyData *)malloc(sizeof(NotifyData) * 1);
+			if(notifyData != NULL)
+			{
+				memset(notifyData,0,sizeof(NotifyData));
+				notifyData->type = PARAM_NOTIFY_RETRY;
+				processNotification(notifyData);
+			}
+			WalPrint("After Sending processNotification\n");
+			c++;
+			if(backoffRetryTime == max_retry_sleep)
+		        {
+	        		c = 3;
+		        	backoffRetryTime = 0;
+			        WalPrint("backoffRetryTime reached max value, reseting to initial value\n");
+		        }
+
+		}
+		else
+		{
+			WalInfo("CMC is equal to 512, cloud and CPE are in sync\n");
+			c = 3;
+			backoffRetryTime = 0;
+			WalPrint("CMC is 512, reseting backoffRetryTime to initial value\n");
+		}
+		WAL_FREE(dbCMC);			
+	}
+	return NULL;
+}
+void SyncNotifyRetryTask()
+{
+	int err = 0;
+	pthread_t threadId;
+	g_syncRetryThreadStarted = 0;
```

Comment:
why this is added here?

Reply:
moved after "SyncNotifyRetry Thread created Successfully"